src/rust/engine/graph/src/lib.rs

```diff
@@ -52,22 +52,71 @@ impl RunToken {
   }
 }
 
+///
+/// A token associated with a Node that is incremented whenever its output value has (or might
+/// have) changed. When a dependent consumes a dependency at a particular generation, that
+/// generation is recorded on the consuming edge, and can later used to determine whether the
+/// inputs to a node have changed.
+///
+/// Unlike the RunToken (which is incremented whenever a node re-runs), the Generation is only
```

Comment:
I'd be tempted to name these `RunGeneration` and `OutputGeneration` or something similarly contrasting.

Reply:
They have very different uses, and aren't really related to one another at that level... so would rather keep the existing separation.